tools/ratbagd.py

```diff
@@ -255,6 +255,10 @@ def __init__(self, api_version):
         result = self._get_dbus_property("Devices") or []
         self._devices = [RatbagdDevice(objpath) for objpath in result]
         self._proxy.connect("notify::g-name-owner", self._on_name_owner_changed)
+        if self.api_version == None and not self._proxy.get_cached_property_names():
```

Comment:
this seems a big backwards, even in the current implementation. we check for `Devices` first, then fail on the API version later. looks like instead of just checking the property names it makes more sense to check `Devices` for `None` and use that before assigning `self._devices`. IOW, reshuffling the order of `__init__` would be a good tidyup.